libs/blocks/merch/merch.js

```diff
@@ -1,243 +1,138 @@
+ * workflow - merch link -> metadata -> default (UCv3)
  * workflowStep - merch link -> default (email)
  * marketSegment - merch link -> default (COM)
- * @param {*} searchParams link level overrides for checkout parameters
+ * @param {{params: URLSearchParams, service: {settings: object}}} context
  * @returns checkout context object required to build a checkout url
  */
-export function getCheckoutContext(searchParams, config) {
-  const { commerce } = config;
-  const checkoutClientId = commerce?.checkoutClientId;
-  const checkoutWorkflow = searchParams.get('checkoutType') ?? getMetadata('checkout-type');
-  const checkoutWorkflowStep = searchParams
+export async function getCheckoutContext(el, params) {
+  const context = await getCommerceContext(el, params);
+  if (!context) return null;
+  const { settings } = await initService();
+  const { checkoutClientId } = settings;
+  const checkoutMarketSegment = params.get('marketSegment');
+  const checkoutWorkflow = params.get('workflow')
+    ?? getMetadata('checkout-workflow')
+    ?? settings.checkoutWorkflow;
+  const checkoutWorkflowStep = params
     ?.get('workflowStep')
-    ?.replace('_', '/');
-  const checkoutMarketSegment = searchParams.get('marketSegment');
-
+    ?.replace('_', '/')
+    ?? settings.checkoutWorkflowStep;
   return {
+    ...context,
     checkoutClientId,
     checkoutWorkflow,
     checkoutWorkflowStep,
     checkoutMarketSegment,
   };
 }
 
-export default async function init(el) {
-  if (!el?.classList?.contains('merch')) return undefined;
-  const fail = await window.tacocat.loadPromise;
-  if (fail) {
-    return undefined;
+export async function getPriceContext(el, params) {
+  const context = await getCommerceContext(el, params);
+  if (!context) return null;
+  const displayOldPrice = context.promotionCode ? params.get('old') : undefined;
+  const displayPerUnit = params.get('seat');
+  const displayRecurrence = params.get('term');
+  const displayTax = params.get('tax');
+  const type = params.get('type');
+  const template = type === 'price' ? undefined : type;
+  return {
+    ...context,
+    displayOldPrice,
+    displayPerUnit,
+    displayRecurrence,
+    displayTax,
+    template,
+  };
+}
+
+export async function buildCta(el, params) {
+  const context = await getCheckoutContext(el, params);
+  if (!context) return null;
+  const service = await initService();
+  const cta = service.createCheckoutLink(
+    context,
+    el.textContent?.replace(/^CTA +/, ''),
+  );
+  cta.classList.add('con-button');
+  if (el.closest('.marquee')) {
+    cta.classList.add('button-l');
   }
-  const { searchParams } = new URL(el.href);
-  const osi = searchParams.get('osi');
-  const type = searchParams.get('type');
-  if (!(osi && type)) {
-    el.textContent = '';
-    el.setAttribute('aria-details', 'Invalid merch block');
-    return undefined;
+  if (el.firstElementChild?.tagName === 'STRONG' || el.parentElement?.tagName === 'STRONG') {
```

Comment:
Let's consider maybe delegating this logic to `decorateButtons()` in [decorate.js].(https://github.com/adobecom/milo/blob/main/libs/utils/decorate.js)

Reply:
@Axelcureno I tried to use `decorateButtons` but apparently in merch block we have slightly differing (simpler) logic. So far, `decorateButtons` can help in adding the `blue` CSS class to merch CTA, but not `button-l` and `con-button` classes for all the cases on the test page ("after" link in PR description).
I propose to keep this as is for now and discuss later, as more CTA variants come.

Reply:
actually `button-l` and `con-button` were our initial assumptions but maybe we should simply use decorateButtons